contributing/README.md

```diff
@@ -5,10 +5,33 @@ See [our contributing guide](../.github/CONTRIBUTING.md) to get started.
 This directory contains documentation intended for anyone interested in
 understanding, and contributing changes to, the Consul codebase.
 
-## Contents
+## Overview
+
+This documentation is organized into the following categories. Each category is 
+either a significant architectural layer, or major functional area of Consul. 
+These documents assume a basic understanding of Consul's feature set, which can
+found in the public [user documentation].
+
+[user documentation]: https://www.consul.io/docs
+
+![Overview](./overview.svg)
```

Comment:
@dnephin: There's a typo in this image: `Service Mesh` is spelled as `Sercice Mesh`

Reply:
Good spot! I pushed a fix